app/change_set_persisters/change_set_persister/cleanup_files.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+class ChangeSetPersister
+  class CleanupFiles
+    attr_reader :resource
+    def initialize(change_set_persister: nil, change_set:, post_save_resource: nil)
+      @resource = change_set.resource
+    end
+
+    def run
+      return unless resource.is_a?(FileSet)
+      CleanupFilesJob.perform_later(file_identifiers: identifiers_to_remove)
+    end
+
+    private
+
+      # Returns the file identifiers for the original file and
+      # each derivative attached to a file set.
+      def identifiers_to_remove
+        identifiers = []
+        files = resource.file_metadata
```

Comment:
Can this just be
```ruby
resource.file_metadata.flat_map do |file|
  file.try(:file_identifiers)
end.compact.map(&:to_s)
```
? If that's more readable, at least. Out of curiosity - why the try? Do we ever have things in file metadata that don't respond to file_identifiers?

If not, then

```ruby
resource.file_metadata.flat_map(&:file_identifiers).compact.map(&:to_s)
```